sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -256,3 +282,13 @@ def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] =
         token_info = cast(SupportsTokenInfo, super()).get_token_info(*scopes, options=options)
         within_dac.set(False)
         return token_info
+
+
+def _get_broker_credential() -> Optional[Type["InteractiveBrowserBrokerCredential"]]:
+    # Get the broker credential if available
+    try:
+        from azure.identity.broker import InteractiveBrowserBrokerCredential
```

Comment:
We need to double check the behavior of get_token in the brokered credential.

Does it raise AuthFail or AuthUnavailable?

Reply:
If within DAC, the broker credential token request methods will raise `CredentialUnavailableError`. However, I am noticing that if silent auth fails, we fall back to interactive sign-in. If we want this to be purely silent auth, we'd likely have to introduce some kwarg to signal the credential not to try interactive auth. WDYT?